wgpu-core/src/logging/subscriber/chrome.rs

```diff
@@ -0,0 +1,67 @@
+use smallvec::SmallVec;
+use std::fmt;
+use tracing::field::{Field, Visit};
+use tracing::{Event, Level, Subscriber};
+use tracing_subscriber::{layer::Context, registry::LookupSpan, Layer};
+
+#[derive(Debug, Default)]
+struct FmtEventVisitor {
+    message: String,
+}
+
+impl Visit for FmtEventVisitor {
+    fn record_debug(&mut self, field: &Field, value: &dyn fmt::Debug) {
+        match field.name() {
+            "message" => self.message = format!("{:?}", value),
+            _ => {}
+        }
+    }
+}
+
+pub struct FmtLayer;
+
+impl<S> Layer<S> for FmtLayer
+where
+    S: Subscriber + for<'span> LookupSpan<'span>,
+{
+    fn on_event(&self, event: &Event<'_>, ctx: Context<'_, S>) {
+        let time = chrono::Local::now().format("%Y-%m-%d %H-%M-%S%.6f");
+
+        let mut visitor = FmtEventVisitor::default();
+        event.record(&mut visitor);
+
+        let mut spans: SmallVec<[&str; 8]> = SmallVec::new();
+        for span in ctx.scope() {
+            spans.push(span.name());
+        }
+        let span_string = spans.join(" | ");
+
+        let metadata = event.metadata();
+        let level = match *metadata.level() {
+            Level::ERROR => "ERROR",
+            Level::WARN => "WARN",
+            Level::INFO => "INFO",
+            Level::DEBUG => "DEBUG",
+            Level::TRACE => "TRACE",
+        };
+
+        match *metadata.level() {
+            Level::ERROR | Level::WARN => eprintln!(
```

Comment:
Somewhat unfortunate to duplicate this. I suppose doing a `format!` call is undesirable since it's `String`-allocating?
We could do `format_args!` and then pick the stderr or stdout for a single [write](https://doc.rust-lang.org/stable/std/fmt/fn.write.html) call

Reply:
fmt::write takes fmt::Write which isn't really implemented by much, so I should be able to manage to turn this into a single `write!` call maybe...